### PR TITLE
phoon: update version, add livecheck

### PR DESCRIPTION
**Pull request by samford**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This PR modifies the `phoon` formula to use a date-based version scheme, which corresponds to the only version information that the first-party website provides (the date in the archive file name) and aligns Homebrew with how [other package managers](https://repology.org/project/phoon/versions) handle these versions. This allows us to create a `livecheck` block for `phoon`, as it wouldn't work properly otherwise with the formula's strange `04A` version.

For context, the original PR to add the formula dates back to 2013-12-18 (Homebrew/legacy-homebrew#25316) and the `version` was set to `03A` at that time. If you check the closest [archive.org snapshot](https://web.archive.org/web/20140208024651/https://www.acme.com/software/phoon/) after this date (2014-02-08), the homepage doesn't mention anything about `03A` and the only version information comes from the file name (`phoon_29jun2005.tar.gz` at the time).

If you go back to the [first available snapshot on archive.org](https://web.archive.org/web/19970124211428/https://www.acme.com/software/phoon/) (1997-01-24), the archive file name is `phoon_01apr95.tar.Z`. The current archive file name is `phoon_14Aug2014.tar.gz`, so it seems like `phoon` has used an implicit date-based version scheme for as far back as we can see. The format has changed a bit over time but all the formats are parseable by `Date#parse`.

It's unclear where the `03A` version came from initially (not `phoon`'s website) but Homebrew is the only package manager using this version scheme. Additionally, when `phoon` was updated to `20140814` in Homebrew/legacy-homebrew#41383, the PR author was unsure what to do with the version, so they used `04A` and asked whether they should use `14Aug2014` instead. `04A` was accepted as "consistent with the previous versioning", despite upstream not using this version format.

Long story short, the `03A`/`04A` version scheme has been used in this formula from the beginning but upstream seems to have always used a date-based format. In this type of scenario, we use a `YYYYMMDD` version in formulae and this PR brings `phoon` in line.

### CODE REVIEW

**Comment by samford** [comment]
livecheck seems to consistently fail only on Ubuntu with the following error:

```
==> brew livecheck phoon --full-name --debug
==> FAILED
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/phoon.rb

Formula:          phoon
Livecheckable?:   Yes

URL (homepage):   https://www.acme.com/software/phoon/
Strategy:         PageMatch
Regex:            /href=.*?phoon[._-]v?(\d{1,2}[a-z]+\d{2,4})\.t/i
/usr/bin/curl --disable --cookie-jar /dev/null --globoff --show-error --user-agent Linuxbrew/3.2.12-56-g5e42c2e\ \(Linux\;\ x86_64\ Ubuntu\ 16.04.7\ LTS\)\ curl/7.47.0 --header Accept-Language:\ en --retry 3 -V
/usr/bin/curl --disable --cookie-jar /dev/null --globoff --show-error --user-agent Linuxbrew/3.2.12-56-g5e42c2e\ \(Linux\;\ x86_64\ Ubuntu\ 16.04.7\ LTS\)\ curl/7.47.0 --header Accept-Language:\ en --retry 3 -V
Error: phoon: Unable to get versions
curl: (56) GnuTLS recv error (-110): The TLS connection was non-properly terminated.
```

**Review by SMillerDev — APPROVED** [review]
